include/tmp/file

```diff
@@ -117,60 +103,23 @@ public:
                    std::string_view label = "",
                    std::string_view extension = "");
 
-  /// Returns an implementation-defined handle to this entry
-  /// @returns The underlying implementation-defined handle
-  native_handle_type native_handle() const noexcept;
-
-  /// Returns the size of this file
-  /// @returns the size of this file, in bytes
-  /// @throws std::filesystem::filesystem_error if cannot get a file size
-  std::uintmax_t size() const;
-
-  /// Returns the size of this file
+  /// Moves the managed path recursively to a given target, releasing
+  /// ownership of the managed path; behaves like `std::filesystem::rename`
+  /// even when moving between filesystems
+  /// @note The target path parent is created when this function is called
+  /// @param[in] to A path to the target file or directory
+  /// @throws std::filesystem::filesystem_error if cannot move the owned path
+  void move(const std::filesystem::path& to);
+
+  /// Moves the managed path recursively to a given target, releasing
+  /// ownership of the managed path; behaves like `std::filesystem::rename`
+  /// even when moving between filesystems
+  /// @note The target path parent is created when this function is called
+  /// @param[in]  to A path to the target file or directory
   /// @param[out] ec Parameter for error reporting
-  /// @returns the size of this file, in bytes
-  std::uintmax_t size(std::error_code& ec) const noexcept;
+  void move(const std::filesystem::path& to, std::error_code& ec);
```

Comment:
`std::ofstream` provides a protected method `move` so it is safe to provide a public method `move` for `tmp::file`

However, without explicit declarations of `file::move` calls to it are ambiguous between `entry::move` and `ios_base::move`